### PR TITLE
Questions

### PR DESCRIPTION
**Pull request by Digambar1411**
feat : Added question page.

contexts:
theme
category
question

### CODE REVIEW

File: src/frontend/components/footer/footer.js [hunk 1]
```diff
@@ -17,23 +20,23 @@ export function Footer() {
                     <a href="https://github.com/Digambar1411" target="_blank"  rel="noopener noreferrer">
                         <img
                             className="social-icon"
-                            src={ lightGithubLogo }
+                            src={ theme === "light" ? lightGithubLogo : githubDark}
```
Comment (harshalini) [inline src/frontend/components/footer/footer.js:23]: Liked the use of theme here

File: src/frontend/contexts/question-context.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    const getData = async ()=>{
+        const response =  await axios.get("api/quizzes");
+        setQuestions(response.data.quizes);
+        console.log(response.data.quizes);
+    }
+
+    const getSelectedCategoryQuestions = (name)=>{
+        setCurqueIndex(initialIndexValue);  //if curqueIndex is not reseted then for other category, quiz will start from last categories current index question. hence reseting index to 0
+        navigate("/rules");
+        const getCategoryQuestions = ()=> questions.filter(question=>question.categoryName===name.toLowerCase())
```
Comment (harshalini) [inline src/frontend/contexts/question-context.js:44]: Great! Use of filter for selected category questions

**Review by harshalini — APPROVED** [review]
Overall code is nice and understandable, go ahead to merge

**Review by partha8 — APPROVED** [review]
The code is clean and readable. Go ahead with the merge.

**Comment by Digambar1411** [comment]
> The code is clean and readable. Go ahead with the merge.

thanks for review

File: src/frontend/contexts/question-context.js (hunk 2 above)
Comment (Digambar1411) [reply src/frontend/contexts/question-context.js:44]: thank you

**Comment by Digambar1411** [comment]
> Overall code is nice and understandable, go ahead to merge

thanks for review

**Pull request merged**